model.py

```diff
@@ -9058,11 +9082,31 @@ class ExternalIntegration(Base):
         "ConfigurationSetting", backref="external_integration",
         lazy="joined", cascade="save-update, merge, delete, delete-orphan",
     )
-    
+
+    def __repr__(self):
+        return u"<ExternalIntegration: protocol=%s goal='%s' settings=%d ID=%d>" % (
+            self.protocol, self.goal, len(self.settings), self.id)
+
     @classmethod
-    def lookup(cls, _db, protocol, goal=None):
-        integration = get_one(_db, cls, protocol=protocol, goal=goal)
-        return integration
+    def lookup(cls, _db, protocol, goal, library=None):
+        integrations = _db.query(cls).filter(
+            cls.protocol==protocol, cls.goal==goal
+        ).all()
+
+        if library:
+            integrations = filter(lambda i: library in i.libraries, integrations)
```

Comment:
Can you join the query against cls.libraries and filter then, rather than filtering after the query has run?